pkg/http/api.go

```diff
@@ -394,7 +394,7 @@ func (a *api) onBulkGetState(reqCtx *fasthttp.RequestCtx) {
 func (a *api) getStateStoreWithRequestValidation(reqCtx *fasthttp.RequestCtx) (state.Store, error) {
 	if a.stateStores == nil || len(a.stateStores) == 0 {
 		msg := NewErrorResponse("ERR_STATE_STORES_NOT_CONFIGURED", messages.ErrStateStoresNotConfigured)
-		respondWithError(reqCtx, fasthttp.StatusBadRequest, msg)
+		respondWithError(reqCtx, fasthttp.StatusInternalServerError, msg)
```

Comment:
What does `len(a.stateStores) == 0` guard against?

Put another way - if I have a state store `coolstore` configured and try to send a request to read from `verycoolstore` is that a 400? What if you take away `verycoolstore`?

In general life makes more sense if there's nothing a *client* can do to trigger a 500.

Reply:
Can the client remove a stateStore? I thought the list could only be modified by applying config?

Reply:
> Can the client remove a stateStore? I thought the list could only be modified by applying config?

I'm talking about modifying config.

So imagine I deploy the app with `coolstatestore` and then try to read data from `verycoolstore`?

Then imagine I redeploy the app with no state stores configured in Dapr and try to read data from `verycoolstore`?

What are the results?

Reply:
> In general life makes more sense if there's nothing a client can do to trigger a 500.

What I'm referring to is whether specifying an *unknown* state store results in a 500.

Reply:
> > In general life makes more sense if there's nothing a client can do to trigger a 500.
> 
> What I'm referring to is whether specifying an _unknown_ state store results in a 500.

So, if you have at least 1 statestore configurated, and the client provides an unknown state store, you get a 400. If you have no state stores configured at all, you get a 500.

Reply:
Here is the test that proves the behavior here:
https://github.com/dapr/dapr/blob/master/pkg/http/api_test.go#L1953-L1965